easybuild/easyblocks/l/lammps.py

```diff
@@ -0,0 +1,359 @@
+from easybuild.tools.build_log import EasyBuildError, print_warning, print_msg
+from easybuild.tools.config import build_option
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+
+KOKKOS_CPU_ARCH_LIST = [
+    'ARMv80',  # ARMv8.0 Compatible CPU
+    'ARMv81',  # ARMv8.1 Compatible CPU
+    'ARMv8-ThunderX',  # ARMv8 Cavium ThunderX CPU
+    'BGQ',  # IBM Blue Gene/Q CPUs
+    'Power8',  # IBM POWER8 CPUs
+    'Power9',  # IBM POWER9 CPUs
+    'SNB',  # Intel Sandy/Ivy Bridge CPUs
+    'HSW',  # Intel Haswell CPUs
+    'BDW',  # Intel Broadwell Xeon E-class CPUs
+    'SKX',  # Intel Sky Lake Xeon E-class HPC CPUs (AVX512)
+    'KNC',  # Intel Knights Corner Xeon Phi
+    'KNL',  # Intel Knights Landing Xeon Phi
+]
+
+KOKKOS_CPU_MAPPING = {
+    'sandybridge': 'SNB',
+    'ivybridge': 'SNB',
+    'haswell': 'HSW',
+    'broadwell': 'BDW',
+    'skylake_avx512': 'SKX',
+    'knights-landing': 'KNL',
+}
+
+
+KOKKOS_GPU_ARCH_TABLE = {
+    "3.0": "Kepler30",  # NVIDIA Kepler generation CC 3.0
+    "3.2": "Kepler32",  # NVIDIA Kepler generation CC 3.2
+    "3.5": "Kepler35",  # NVIDIA Kepler generation CC 3.5
+    "3.7": "Kepler37",  # NVIDIA Kepler generation CC 3.7
+    "5.0": "Maxwell50",  # NVIDIA Maxwell generation CC 5.0
+    "5.2": "Maxwell52",  # NVIDIA Maxwell generation CC 5.2
+    "5.3": "Maxwell53",  # NVIDIA Maxwell generation CC 5.3
+    "6.0": "Pascal60",  # NVIDIA Pascal generation CC 6.0
+    "6.1": "Pascal61",  # NVIDIA Pascal generation CC 6.1
+    "7.0": "Volta70",  # NVIDIA Volta generation CC 7.0
+    "7.2": "Volta72",  # NVIDIA Volta generation CC 7.2
+    "7.5": "Turing75",  # NVIDIA Turing generation CC 7.5
+}
+
+
+class EB_LAMMPS(CMakeMake):
+    """
+    Support for building and installing LAMMPS
+    """
+
+    @staticmethod
+    def extra_options(**kwargs):
+        """Custom easyconfig parameters for LAMMPS"""
+
+        extra_vars = {
+            # see https://developer.nvidia.com/cuda-gpus
+            'cuda_compute_capabilities': [[], "List of CUDA compute capabilities to build with", CUSTOM],
+            'general_packages': [None, "List of general packages without `PKG_` prefix.", MANDATORY],
+            'kokkos': [True, "Enable kokkos build.", CUSTOM],
+            'kokkos_arch': [None, "Set kokkos processor arch manually, if auto-detection doesn't work.", CUSTOM],
+            'user_packages': [None, "List user packages without `PKG_USER-` prefix.", MANDATORY],
+        }
+        return CMakeMake.extra_options(extra_vars)
+
+    def prepare_step(self):
+        super(EB_LAMMPS, self).prepare_step()
+
+        # Unset LIBS when using both KOKKOS and CUDA - it will mix lib paths otherwise
+        if self.cfg['kokkos'] and get_software_root('CUDA'):
+            run_cmd("unset LIBS")
```

Comment:
This won't work, you need to use `env.unset_env_vars['LIBS']`.

The commands specified in `run_cmd` are run in a temporary subshell, so you're not actually unsetting `$LIBS` at all here...